### ISSUE TITLE
[FRAME] Purge invalid `Balances::Locks` items

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#1523]
Follow up investigation of https://github.com/paritytech/polkadot-sdk/issues/1149 showed that `Balances::Locks` prefix contains un-decodable keys.  
These (K, V) pairs are basically inaccessible since `translate` and `iter_keys` skip them, but should still be cleaned up.  
Opening an issue since runtime changes are on hold until we have the runtimes repo.

**Comment by ggwpez** [comment]
It was introduced in this block: https://kusama.subscan.io/block/201600  
Can be replayed with `npx @acala-network/chopsticks@latest run-block --endpoint=wss://kusama-rpc.polkadot.io --html -b 0xb428f058b81ed1c058566f00138f2b241b2e706b9bddf293db47adf08eefe586`.

**Comment by liamaharon** [comment]
What's best practice for purging undecodable keys?

**Comment by ggwpez** [comment]
I dont think we have any best practice in this case, since the keys are udnecodable. Otherwise we can just do `iter_keys` but not here.  
So you probably have to resort to `next_kex` from `sp-io` or add some kind of `try_iter_keys` that returns the undecodable ones as well.
